stan/math/prim/mat/fun/matrix_exp_2x2.hpp

```diff
@@ -18,23 +15,27 @@ namespace stan {
          * @return Matrix exponential of A.
          */
         template <typename T>
-        Matrix<T, Dynamic, Dynamic>
-        matrix_exp_compute_2x2(const Matrix<T, Dynamic, Dynamic>& A) {
-            Matrix<T, Dynamic, Dynamic> B;
+        Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
+        matrix_exp_2x2(const Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>& A) {
             T a = A(0, 0), b = A(0, 1), c = A(1, 0), d = A(1, 1), delta;
-            delta = sqrt((a - d)*(a - d) + 4*b*c);
-
-            B.resize(2, 2);
-            B(0, 0) = exp(.5*(a+d))
-                       * (delta*cosh(.5*delta) + (a-d)*sinh(.5*delta));
-            B(0, 1) = 2*b*exp(.5*(a+d)) * sinh(.5*delta);
-            B(1, 0) = 2*c*exp(.5*(a+d)) * sinh(.5*delta);
-            B(1, 1) = exp(.5*(a+d))
-                       * (delta*cosh(.5*delta) + (d-a)*sinh(.5*delta));
-
-            B = 1/delta * B;
-
-            return B;
+            delta = sqrt((a - d) * (a - d) + 4 * b * c);
+
+			Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> B(2, 2);
+			T half_delta = 0.5 * delta;
+			T cosh_half_delta = cosh(half_delta);
+			T sinh_half_delta = sinh(half_delta);
+			T exp_half_a_plus_d = exp(0.5 * (a + d));
+			T exp_sinh = exp_half_a_plus_d * sinh_half_delta;
+			T delta_cosh = delta * cosh_half_delta;
+			
+			B(0, 0) = exp_half_a_plus_d * (delta_cosh + 
+					  (a - d) * sinh_half_delta);
```

Comment:
Sorry to be so pedantic on style, but operators should start lines, not end lines and terms should be kept to gether to the extent possible sot hese are easy to read.

Also, you can save `(a - d) * sinh_half_delta` because the `(d - a) * sinh_half_delta` below can just be subtracted.